ui/src/app/base/sagas/actions.ts

```diff
@@ -137,8 +143,68 @@ export function* updateDomainRecord(
   );
 }
 
+/**
+ * Generate action to call after the initial DNS resource delete action is
+ * complete.
+ * @param params - The params for the domain/deleteRecord action.
+ * @returns The next action to call.
+ */
+export const generateNextDeleteRecordAction = (
+  params: DeleteRecordParams
+): (() => WebSocketAction)[] => {
+  const { deleteResource, domain, rrset } = params;
+  return deleteResource
```

Comment:
Same here, might be good to leave a comment about when/why this next action is required.